internal/cache/cache.go

```diff
@@ -25,21 +27,44 @@ type FileCache[T any] struct {
 	key            string
 }
 
-func NewFileCache[T any](ctx context.Context, key string, dur time.Duration) (*FileCache[T], error) {
+type CacheSettings struct {
+	Key               string
+	Namespace         string
+	ClearOnNewVersion bool
+	Duration          time.Duration
+}
+
+func NewFileCache[T any](ctx context.Context, settings CacheSettings) (*FileCache[T], error) {
 	home, err := os.UserHomeDir()
 	if err != nil {
 		return nil, err
 	}
 
 	cfgDir := filepath.Join(home, ".speakeasy", "cache")
-	DeleteOldCache(cfgDir, key, dur)
+	DeleteOldCache(cfgDir, settings.Namespace, settings.Duration)
+	builder := strings.Builder{}
```

Comment:
Nit: can we name this what thing we're building rather than `builder`? "Builder" would be like naming a variable "string" 😆